src/curl.c

```diff
@@ -180,7 +188,7 @@ curl_multi_sock_cb(CURL *easy, curl_socket_t fd, int what, void *envp,
 	struct curl_env *env = (struct curl_env *) envp;
 	struct ev_io *watcher = (struct ev_io *) watcherp;
 
-	if (what == CURL_POLL_REMOVE) {
+	if (what == CURL_POLL_REMOVE || env->cleanup == 1) {
```

Comment:
Is it possible to run several chunked requests?

I guess if we have two such requests behaviour will be following:
- gc sets cleanup - true
- the first finished request removes mempool
- application crashes because the second request is in fly

An approach with fair refcounts should be better. At first it considers the case above. At second you don't need to handle cleanup == 1 case in different places since it makes code more complex and unclear